tests/unit/algo/anomaly/test_openvino_model.py

```diff
@@ -0,0 +1,50 @@
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+
+import pytest
+import torch
+from anomalib.metrics.min_max import MinMax
+from anomalib.metrics.threshold import ManualThreshold
+from otx.algo.anomaly.openvino_model import AnomalyOpenVINO
+from otx.algo.anomaly.padim import Padim
+from otx.algo.anomaly.stfpm import Stfpm
+from otx.core.types.export import OTXExportFormatType
+from otx.core.types.label import AnomalyLabelInfo
+from otx.core.types.task import OTXTaskType
+
+
+class TestAnomalyOpenVINO:
+    @pytest.fixture(
+        params=[
+            # "padim", Cannot export padim in this way. We will face an error.
+            "stfpm",
+        ],
+    )
+    def fxt_ov_ir(self, request, tmpdir):
+        if request.param == "padim":
+            model = Padim()
+        elif request.param == "stfpm":
+            model = Stfpm()
+        else:
+            raise ValueError
+
+        # NOTE: if we do not inject those into `model`,
+        # we will face errors during `model.export()` such as:
+        # AttributeError: 'Stfpm' object has no attribute 'normalization_metrics'
+        model.normalization_metrics = MinMax()
+        model.normalization_metrics.min = torch.tensor(-1.0)
+        model.normalization_metrics.max = torch.tensor(1.0)
+        model.image_threshold = ManualThreshold(0.0)
+        model.pixel_threshold = ManualThreshold(0.0)
+        model.task = OTXTaskType.ANOMALY_CLASSIFICATION
```

Comment:
```python
model = Padim()
model.export(
            output_dir=tmpdir,
            base_name="exported_model",
            export_format=OTXExportFormatType.OPENVINO,
        )
```

Hi @ashwinvaidya17,
It seems impossible to export OTX-integrated anomalib models in this way. Can we enhance this?

Reply:
If not, is there anyway to enhance just this test code neater without injecting those variables explicitly?

Reply:
These variables are set from the callbacks when the `setup` method is called in the callback.
https://github.com/openvinotoolkit/training_extensions/blob/80acb86508c12d5602eb93399a91a476b3feffd3/src/otx/core/model/anomaly.py#L230

One option would be to assign default values here https://github.com/openvinotoolkit/training_extensions/blob/80acb86508c12d5602eb93399a91a476b3feffd3/src/otx/core/model/anomaly.py#L149 so that export works when the model is created. But I don't know how it will behave with the callbacks. And, since the default values are incorrect, directly calling export on a new model will lead to an incorrect OV model.

Reply:
Yeah that would be better